plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java

```diff
@@ -151,9 +153,9 @@
             .withChronology(IsoChronology.INSTANCE);
 
     @Inject
-    public DruidJdbcClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, IdentifierMapping identifierMapping, RemoteQueryModifier queryModifier)
+    public DruidJdbcClient(BaseJdbcConfig config, MappingConfig mappingConfig, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, IdentifierMapping identifierMapping, RemoteQueryModifier queryModifier)
     {
-        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, queryModifier, false);
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), mappingConfig.isCaseInsensitiveNameMatching() ? identifierMapping : new DefaultDruidIdentifierMapping(), queryModifier, false);
```

Comment:
this condition appears reversed?

Does applying this diff locally on top of this PR also solve the issue you're trying to address?

```diff
diff --git a/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java b/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java
index 72f1bee1fc..f1e0fd69ef 100644
--- a/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java
+++ b/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java
@@ -108,7 +108,7 @@ public abstract class BaseJdbcClient
     protected final String identifierQuote;
     protected final Set<String> jdbcTypesMappedToVarchar;
     protected final RemoteQueryModifier queryModifier;
-    protected IdentifierMapping identifierMapping;
+    protected final IdentifierMapping identifierMapping;
     private final boolean supportsRetries;
 
     public BaseJdbcClient(
diff --git a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DefaultDruidIdentifierMapping.java b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidIdentifierMapping.java
similarity index 60%
rename from plugin/trino-druid/src/main/java/io/trino/plugin/druid/DefaultDruidIdentifierMapping.java
rename to plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidIdentifierMapping.java
index 155a5622c5..32c6d72854 100644
--- a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DefaultDruidIdentifierMapping.java
+++ b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidIdentifierMapping.java
@@ -14,19 +14,23 @@
 package io.trino.plugin.druid;
 
 import io.trino.plugin.jdbc.mapping.DefaultIdentifierMapping;
+import io.trino.plugin.jdbc.mapping.ForwardingIdentifierMapping;
+import io.trino.plugin.jdbc.mapping.IdentifierMapping;
 
 import java.sql.Connection;
 
-/*
- * Identifier mapping when case insensitive name matching is disabled
- */
-public class DefaultDruidIdentifierMapping
-        extends DefaultIdentifierMapping
+public class DruidIdentifierMapping
+        extends ForwardingIdentifierMapping
 {
     @Override
-    protected boolean storesUpperCaseIdentifiers(Connection connection)
+    protected IdentifierMapping delegate()
     {
-        // false since the Druid driver incorrectly returns true for this always
-        return false;
+        return new DefaultIdentifierMapping() {
+            @Override
+            protected boolean storesUpperCaseIdentifiers(Connection connection)
+            {
+                return false;
+            }
+        };
     }
 }
diff --git a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java
index 839e4698b0..62fec3d405 100644
--- a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java
+++ b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClient.java
@@ -153,9 +153,9 @@ public class DruidJdbcClient
             .withChronology(IsoChronology.INSTANCE);
 
     @Inject
-    public DruidJdbcClient(BaseJdbcConfig config, MappingConfig mappingConfig, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, IdentifierMapping identifierMapping, RemoteQueryModifier queryModifier)
+    public DruidJdbcClient(BaseJdbcConfig config, ConnectionFactory connectionFactory, QueryBuilder queryBuilder, IdentifierMapping identifierMapping, RemoteQueryModifier queryModifier)
     {
-        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), mappingConfig.isCaseInsensitiveNameMatching() ? identifierMapping : new DefaultDruidIdentifierMapping(), queryModifier, false);
+        super("\"", connectionFactory, queryBuilder, config.getJdbcTypesMappedToVarchar(), identifierMapping, queryModifier, false);
     }
 
     @Override
diff --git a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClientModule.java b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClientModule.java
index 6dd2d378f6..b52bb55e3a 100644
--- a/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClientModule.java
+++ b/plugin/trino-druid/src/main/java/io/trino/plugin/druid/DruidJdbcClientModule.java
@@ -38,6 +38,7 @@ public class DruidJdbcClientModule
     @Override
     public void configure(Binder binder)
     {
+        binder.bind(DruidIdentifierMapping.class).in(Scopes.SINGLETON);
         binder.bind(JdbcClient.class).annotatedWith(ForBaseJdbc.class).to(DruidJdbcClient.class).in(Scopes.SINGLETON);
         newSetBinder(binder, ConnectorTableFunction.class).addBinding().toProvider(Query.class).in(Scopes.SINGLETON);
     }
```

Reply:
@kokosing can you take a look as well to confirm I'm not mistunderstanding this PR.

The issue is that the normal `DefaultIdentifierMapping` can't be used with Druid because Druid JDBC driver returns incorrect value for `storesUppercaseIdentifier`.

Reply:
I think this condition should be handled in guice binding.

Reply:
@hashhar the DefaultIdentifierModule is bound here by default https://github.com/trinodb/trino/blob/master/plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/mapping/IdentifierMappingModule.java#L57 when the case-insensitive-matching flag is false. So if you check `TestDruidConnectorTest` there are failures because thats when `DefaultIdentifierMapping` is called. 

That said, I dont know if there is a better way to do this via guice instead of the `if` logic I have here. With your diff, the identifierMapping is still `DefaultIdentifierMapping` (when case-insensitive-matching=false)